api/datastore/internal/datastoreutil/metrics.go

```diff
@@ -104,7 +106,24 @@ func (m *metricds) GetCalls(ctx context.Context, filter *models.CallFilter) ([]*
 func (m *metricds) InsertLog(ctx context.Context, appName, callID string, callLog io.Reader) error {
 	span, ctx := opentracing.StartSpanFromContext(ctx, "ds_insert_log")
 	defer span.Finish()
-	return m.ds.InsertLog(ctx, appName, callID, callLog)
+
+	var b bytes.Buffer
+	gzippedLog := gzip.NewWriter(&b)
+	defer gzippedLog.Close()
+	_, err := io.Copy(gzippedLog, callLog)
+	if err != nil {
+		return err
+	}
+	err = gzippedLog.Flush()
+	if err != nil {
+		return err
+	}
+	newCallLog, err := gzip.NewReader(&b)
+	if err != nil {
+		return err
+	}
+
+	return m.ds.InsertLog(ctx, appName, callID, newCallLog)
```

Comment:
let's punt on this (didn't get a reply from you). we should discuss. we want to thread gzip through the whole system really, it's not this straightforward. then we aren't gathering un-compressed bytes (in memory/files) and then compressing them just on upload. let's just fix plain text api here

Reply:
metric ds is kind of the wrong place to put this, too.

Reply:
anyway i ran into complications trying that (hence my abandonment). but now i know what was wrong (there was another since fixed bug in logs)

Reply:
I'm not sure i understand your first comment.

So, if that is not metrics ds, so gzip compressing should be just in every logstore implementation, which doesn't seem reasonable.

Reply:
metric ds is not the only option. i don't think we need any implementation. i can dig up my patch and try to explain more concretely, am happy to push forward with the plaintext api meanwhile (based on Accept headers)

Reply:
there's one problem - swagger, not sure that they support multiple kinds of responses in different formation: same response code, but the body is different.

Reply:
i'd like to deal with gzip thing in first place, only then plaintext api thing.

Reply:
we can change swagger to plaintext, that's fine, and new clients will use it, but we need backwards compat for old clients using json still. the gzip isn't pressing, really, and there's code that has to be written for compatibility so it's not perfectly straightforward

Reply:
Done, API remains compatible with older clients. So, i'd like to know your thoughts and how you see gzip'ing function's logs.

As far as i understand there's nothing much code needed to do actual compressing, right? But the bigger question to me is where it happens in order to be the same for all stores (sql and minion at this moment, obviously it should happen just right before going to actual implementations).